### PR TITLE
[ONNX][TOPI] Support select_last_index for argmin/max

### PR DESCRIPTION
**Pull request by AndrewZhaoLuo**
Add support for topi `select_last_index`, which grabs the last occurance of the target element for `argmin` and `argmax`.

Add support for opset13 versions of argmin and argmax which have `select_last_index`. Enable all argmin and argmax onnx tests.

Alternatives:
- A cleaner way is to add an option to all reduction options to reverse the order of traversal when running the reduce. However, after digging deeper I cannot seem to figure out the best way or even a way to do this cleanly.

### CODE REVIEW

**Comment by AndrewZhaoLuo** [comment]
Ah, this is actually quite a bit tricky since for non cpu schedules, the current reduce operations are not stable.

**Comment by AndrewZhaoLuo** [comment]
With some additional conditions on the merge, I believe a stable argmin argmax is made.

**Comment by junrushao1994** [comment]
Would be nice if someone could help review the PR! CC @jcf94 @mbrookhart @jwfromm

File: include/tvm/relay/attrs/reduce.h [hunk 1]
```diff
@@ -61,6 +61,42 @@ struct ReduceAttrs : public tvm::AttrsNode<ReduceAttrs> {
   }
 };
 
+/*! \brief Attributes for Reduce operators which reduce by finding a single element. E.g. argmin */
+struct OneElementReduceAttrs : public tvm::AttrsNode<OneElementReduceAttrs> {
```
Comment (masahi) [inline include/tvm/relay/attrs/reduce.h:65]: I prefer `OneElementReduce` -> `ArgReduce`

File: src/relay/op/tensor/reduce.cc [hunk 2]
```diff
@@ -269,29 +290,46 @@ inline std::vector<IndexExpr> ReduceShapeImpl(const std::vector<IndexExpr>& in_s
...
+ * \brief SingleElementArgReduceRel Output type and shape relation evaluation function.
+ * \param num_inputs Number of input types in the args.
+ * \param attrs The additional attributes of the operator.
+ * \param reporter The reporter to report solution to.
+ * \return false if This relation cannot be resolved. true if this relation has been resolved.
+ */
+bool SingleElementArgReduceRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
+                               const TypeReporter& reporter) {
+  return GenericReduceRel<OneElementReduceAttrs>(types, num_inputs, attrs, reporter);
+}
```
Comment (masahi) [inline src/relay/op/tensor/reduce.cc:332]: No need to introduce `ArgReduceRel` and `SingleElementArgReduceRel` boilerplate. Just use `.add_type_rel("ArgReduce", GenericReduceRel<ReduceAttrs>)`.

File: include/tvm/relay/attrs/reduce.h (hunk 1 above)
Comment (AndrewZhaoLuo) [reply include/tvm/relay/attrs/reduce.h]: Done

File: src/relay/op/tensor/reduce.cc (hunk 2 above)
Comment (AndrewZhaoLuo) [reply src/relay/op/tensor/reduce.cc]: Done

**Comment by AndrewZhaoLuo** [comment]
PTAL @masahi

File: include/tvm/topi/reduction.h [hunk 3]
```diff
@@ -442,35 +481,49 @@ inline Tensor max(const Tensor& data, const Array<Integer>& axis, bool keepdims
...
+    if (select_last_index) {
+      proper_index = lhs_idx > rhs_idx;
+    } else {
+      proper_index = lhs_idx < rhs_idx;
+    }
+
+    PrimExpr update_index = is_bigger || (is_same && proper_index);
+    result.push_back(tvm::tir::Select(update_index, lhs[0], rhs[0]));  // idx
+    result.push_back(tvm::tir::Select(is_bigger, lhs[1], rhs[1]));     // val
+    LOG(WARNING) << result;
```
Comment (mbrookhart) [inline include/tvm/topi/reduction.h:523]: Remove the log? I think this will end up being noisy.

File: src/relay/op/tensor/reduce.cc [hunk 4]
```diff
@@ -207,9 +208,29 @@ Array<te::Tensor> ReduceCompute(const Attrs& attrs, const Array<te::Tensor>& inp
       return {topi::identity(inputs[0])};
     }
   }
+
   return {f(inputs[0], axes, param->keepdims, false)};
 }
 
+template <typename F>
+Array<te::Tensor> OneElementReduceCompute(const Attrs& attrs, const Array<te::Tensor>& inputs,
```
Comment (mbrookhart) [inline src/relay/op/tensor/reduce.cc:216]: I don't understand what OneElement means here?

**Review by mbrookhart — COMMENTED** [review]
One leftover print, one confusion about naming, otherwise looks good

File: include/tvm/topi/reduction.h (hunk 3 above)
Comment (AndrewZhaoLuo) [reply include/tvm/topi/reduction.h]: Good catch, done

File: src/relay/op/tensor/reduce.cc (hunk 4 above)
Comment (AndrewZhaoLuo) [reply src/relay/op/tensor/reduce.cc]: Hmm, I've changed it to ArgReduceCompute to be consistent with the other naming suggested by Masa.

**Review by mbrookhart — APPROVED** [review]
LGTM

**Review by masahi — APPROVED** [review]
Please kick CI again.

**Comment by masahi** [comment]
Thanks @AndrewZhaoLuo @mbrookhart

**Pull request merged**